### ISSUE TITLE
CLion/Ninja incompatible with /MP flag

### ISSUE THREAD

**Issue by LouChiSoft** [RavEngine/RavEngine#11]
Hi, recently found this project and decided to see if I could get it working with CLion on Windows. The good news is that it almost does. I had to make a slight change to the CMake file for RavEngine.

I changed line 97:
`add_definitions(/MP)`
to
```
if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
    add_definitions(/MP)
endif()
```
But once I did that the project builds, runs and debugs just fine. I don't think it should cause too much of an issue. IIRC /MP is for using multiple threads to build which I _think_ Ninja handles automatically by just using multiple processes. Additionally there are a large number of compiler flag warnings `Command line warning D9025 : overriding '/MTd' with '/MDd'` etc. These don't break the build, but possible need looking into

**Comment by Ravbug** [comment]
Thank you for your report! I have incorporated your suggestion: https://github.com/RavEngine/RavEngine/blob/a9536976d19ec9a728779894ca7ccc1ec03b9f01/CMakeLists.txt#L97-L99
Out of curiosity, does CLion + Ninja work with the UWP builds? I don't have a CLion license so I can't try it.

**Comment by LouChiSoft** [comment]
I've never worked with UWP before but if you have a setup guide for it I would be happy to test it for you

**Comment by Ravbug** [comment]
To build for UWP, you should only need to set `CMAKE_SYSTEM_NAME=WindowsStore` and `CMAKE_SYSTEM_VERSION="10.0"` on the command line. 

This project supports UWP (despite it being deprecated) because that's currently the only way to develop for Xbox without signing the GDK NDA and becoming an official Xbox developer.

**Comment by LouChiSoft** [comment]
I see, that's fair enough. I wasn't able to get it to build. With just `CMAKE_SYSTEM_NAME` amd `CMAKE_SYSTEM_VERSION` enabled it wasn't compiling, but I noticed that it also wasn't defining `_WINRT` either for the UWP version `START_APP`. I manually defined it and it still was unable to builod. SDL seems to be the erroring member. 
`cl : Command line error D8048 : cannot compile C file 'C:\Dev\CPP\Planes\RavEngine\deps\SDL2\src\SDL_assert.c' with the /ZW option` for example.
